Jenkinsfile

```diff
@@ -175,6 +175,7 @@ pipeline {
                     steps {
                         timeout(time: longTimeoutMinutes, unit: 'MINUTES') {
                             runMavenContainerCommand('.ci/scripts/distribution/test-java.sh')
+                            runMavenContainerCommand('.ci/scripts/distribution/random-test-java.sh')
```

Comment:
This way the test execution is still part of the same test stage (`Test (Java)`). We could also extract it to another stage for parallel execution, or keeping it sequential but like the `IT` stage providing a visual representation of this separation in the flow. WDYT?

Reply:
Yes, I would like that. I had a different version where it was in a parallel track. But it was a mess because somehow both were running in the same agent/container (never really sure what the right name is). So basically to mavens were running at the same time on the same tree structure and getting in each others way. 

If you know a way to run this in its own environment, I'm all ears.

Reply:
I'm not entirely sure our current system can property deal with more parallelisation (i.e. adding more agents to the flow). But if you wanted to do that have a look at how the `IT` stage uses another agent https://github.com/zeebe-io/zeebe/blob/80d7bc07222e41794dd8ec260fb12a9d46edaf3e/Jenkinsfile#L207-L222.

For now, it might be better to keep it sequential. We can however make the sequence more visible by splitting the maven commands into 2 different stages. To do this you take the `stage('Test (Java)')` and replace the `steps` by the keyword `stages` and to that stages you add a `stage('Unit tests')` and a `stage('Randomised tests')`. Each of those stages than contains the steps that are now in `stage('Test (Java)')`. Does that description make sense? I'm also okay with leaving it implicit as it is. Let me know what you think is best.

Reply:
I would prefer to leave it as it is. The Jenkins file is becoming difficult to work with. Also, the two scripts do almost exactly the same. For the other stages they usually differ in a meaningful way from each other. So, In essence, I would prefer to keep it like that for now.